app/src/main/java/org/oppia/android/app/home/promotedlist/ComingSoonTopicsListView.kt

```diff
@@ -0,0 +1,37 @@
+package org.oppia.android.app.home.promotedlist
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.recyclerview.widget.RecyclerView
+import org.oppia.android.app.recyclerview.BindableAdapter
+import org.oppia.android.app.recyclerview.StartSnapHelper
+import org.oppia.android.databinding.ComingSoonTopicViewBinding
+import org.oppia.android.databinding.PromotedStoryCardBinding
+
+/**
+ * A custom [RecyclerView] for displaying a variable list of promoted lesson stories that snaps to
+ * a fixed position when being scrolled.
+ */
+class ComingSoonTopicsListView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : RecyclerView(context, attrs, defStyleAttr) {
+
+  init {
+    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<ComingSoonTopicsViewModel>()
+      .registerViewDataBinderWithSameModelType(
+        inflateDataBinding = ComingSoonTopicViewBinding::inflate,
+        setViewModel = ComingSoonTopicViewBinding::setViewModel
+      ).build()
+
+    /*
+     * The StartSnapHelper is used to snap between items rather than smooth scrolling, so that
+     * the item is completely visible in [HomeFragment] as soon as learner lifts the finger
+     * after scrolling.
+     */
+    val snapHelper = StartSnapHelper()
```

Comment:
Shouldn't this be done after view has been created? this could be null here if i'm not wrong

Reply:
The snap works on first complete visible item. If this method returns null, SnapHelper will not snap to any view. 
I have to read more about it. If you still have concerns please let me know I will follow up with you on this.

Reply:
I need to check this myself too i would update on this once i'm done with some through testing